exec/totemconfig.c

```diff
@@ -336,16 +338,28 @@ static int totem_volatile_config_validate (
 	}
 
 	if (totem_config->token_retransmit_timeout < MINIMUM_TIMEOUT) {
-		snprintf (local_error_reason, sizeof(local_error_reason),
-			"The token retransmit timeout parameter (%d ms) may not be less than (%d ms).",
-			totem_config->token_retransmit_timeout, MINIMUM_TIMEOUT);
-		goto parse_error;
+		if (icmap_get_uint32("totem.token_retransmit", &tmp_config_value) == CS_OK) {
+			snprintf (local_error_reason, sizeof(local_error_reason),
+				"The token retransmit timeout parameter (%d ms) may not be less than (%d ms).",
+				totem_config->token_retransmit_timeout, MINIMUM_TIMEOUT);
+			goto parse_error;
+		}
+		else {
+			snprintf (local_error_reason, sizeof(local_error_reason),
+					"The token timeout should be at least (%d ms) based on (%d) retransmits count, or retransmits count should be at most (%d) based on token timeout is (%d ms).",
+					(int)(MINIMUM_TIMEOUT*(totem_config->token_retransmits_before_loss_const + 0.2)),
+					totem_config->token_retransmits_before_loss_const,
+					(int)((5*totem_config->token_timeout - MINIMUM_TIMEOUT) / (5*MINIMUM_TIMEOUT)),/*from token_timeout/(count+0.2) >= MINIMUM_TIMEOUT, Make sure get accurate round number*/
```

Comment:
Line is super long, please make it shorter. Also is it really needed to use magic `5*` value? Couldn't be used floats?